docs/Client.md

```diff
@@ -0,0 +1,1011 @@
+    - [`Client.destroy(error)`](#clientdestroyerror)
+      - [Example - Request is aborted when Client is destroyed](#example---request-is-aborted-when-client-is-destroyed)
+    - [`Client.dispatch(options, handlers)`](#clientdispatchoptions-handlers)
+      - [Parameter: `DispatchOptions`](#parameter-dispatchoptions)
+      - [Parameter: `DispatchHandlers`](#parameter-dispatchhandlers)
+      - [Example 1 - Dispatch GET request](#example-1---dispatch-get-request)
+      - [Example 2 - Dispatch Upgrade Request](#example-2---dispatch-upgrade-request)
+    - [`Client.pipeline(options, handler)`](#clientpipelineoptions-handler)
+      - [Parameter: PipelineOptions](#parameter-pipelineoptions)
+      - [Parameter: PipelineHandlerData](#parameter-pipelinehandlerdata)
+      - [Example 1 - Pipeline Echo](#example-1---pipeline-echo)
+    - [`Client.request(options [, callback])`](#clientrequestoptions--callback)
+      - [Parameter: `RequestOptions`](#parameter-requestoptions)
+      - [Parameter: `ResponseData`](#parameter-responsedata)
+      - [Example 1 - Basic GET Request](#example-1---basic-get-request)
+      - [Example 2 - Aborting a request](#example-2---aborting-a-request)
+    - [`Client.stream(options, factory [, callback])`](#clientstreamoptions-factory--callback)
+      - [Parameter: `StreamFactoryData`](#parameter-streamfactorydata)
+      - [Parameter: `StreamData`](#parameter-streamdata)
+      - [Example 1 - Basic GET stream request](#example-1---basic-get-stream-request)
+      - [Example 2 - Stream to Fastify Response](#example-2---stream-to-fastify-response)
+    - [`Client.upgrade(options[, callback])`](#clientupgradeoptions-callback)
+      - [Parameter: `UpgradeOptions`](#parameter-upgradeoptions)
+      - [Parameter: `UpgradeData`](#parameter-upgradedata)
+      - [Example 1 - Basic Upgrade Request](#example-1---basic-upgrade-request)
+  - [Instance Properties](#instance-properties)
+    - [`Client.busy`](#clientbusy)
+    - [`Client.closed`](#clientclosed)
+    - [`Client.connected`](#clientconnected)
+    - [`Client.destroyed`](#clientdestroyed)
+    - [`Client.pending`](#clientpending)
+    - [`Client.pipelining`](#clientpipelining)
+    - [`Client.running`](#clientrunning)
+    - [`Client.size`](#clientsize)
+  - [Instance Events](#instance-events)
+    - [Event: `'connect'`](#event-connect)
+      - [Example - Client connect event](#example---client-connect-event)
+    - [Event: `'disconnect'`](#event-disconnect)
+      - [Example - Client disconnect event](#example---client-disconnect-event)
+    - [Event: `'drain'`](#event-drain)
+      - [Example - Client drain event](#example---client-drain-event)
+  - [Parameter: `UndiciHeaders`](#parameter-undiciheaders)
+    - [Example 1 - Object](#example-1---object)
+    - [Example 2 - Array](#example-2---array)
+
+## `new Client(url, [options])`
+
+Arguments:
+
+* **url** `URL | string` - It should only include the protocol, hostname, and port.
```

Comment:
The `It should only include the protocol, hostname, and port.` is often missed. Can we somehow emphasize it? Maybe also explain why and provide a link to `undici.request` etc... methods

Reply:
Yeah we can emphasize this just by bolding that text so its more obvious. For further explanations and other options I think that should belong in an example section or in another document. This page is primarily focussed on being an API doc for those who want the core details without reading the source code.